unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Audio/AudioContainer.cs

```diff
@@ -29,9 +30,32 @@ void Awake()
             maxDistance = 400f;
         }
 
+        if (instantiateEvents)
+        {
+            for(int i = 0; i < audioEvents.Count; i++)
+            {
+                string str = audioEvents[i].name;
+                AudioEvent instance = Instantiate(audioEvents[i]);
+                audioEvents[i] = instance;
+                instance.name = str;
+            }
+        }
+
         foreach (AudioEvent e in audioEvents)
         {
             e.Initialize(this);
         }
     }
+
+    public AudioEvent GetEvent(string eventName)
+    {
+        for (int i = 0; i < audioEvents.Count; i++)
+        {
+            if (audioEvents[i].name == eventName)
+                return audioEvents[i];
+        }
+
+        Debug.Log(name + "'s AudioContainer couldn't find an event called " + eventName);
+        return null;
+    }
```

Comment:
Unless the audioEvents list is changing on runtime, we could create a dictionary <audioEvent.Name, AudioEvent> from it in the Awake. This would increase performance and reduce the boilerplate needed to locate an audioEvent.

Reply:
Ah, this is a good idea. The list won't change during runtime.

Reply:
Looking into this, it looks like Dictionaries aren't available in the inspector by default. Would have to write a custom inspector. Will definitely look into this at some point later.